src/hooks/useTodos.js

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react';
+import { createTodo, fetchTodos } from '../services/client.js';
+
+export function useTodos() {
+
+  const [todos, setTodos] = useState([]);
+    
+  useEffect(() => {
+    async function getTodos() {
+      const data = await fetchTodos();
+      setTodos(data);
+    }
+    getTodos();
+  }, []);
+
+  useEffect(() => {
+    async function addTodo() {
+      const data = await createTodo();
+      setTodos(data);
+    }
+    addTodo();
+  }, []);
```

Comment:
I'm not sure if this useEffect is necessary and it may be causing bugs (note behavior if you refresh the page while looking at a list of your todos). You already have a useEffect that will get all your todos on page load, but then this useEffect is also being called on page load, but it creates a todo?